p2p/net/conn/dial.go

```diff
@@ -26,9 +27,13 @@ func (d *Dialer) String() string {
 // Ensures raddr is part of peer.Addresses()
 // Example: d.DialAddr(ctx, peer.Addresses()[0], peer)
 func (d *Dialer) Dial(ctx context.Context, raddr ma.Multiaddr, remote peer.ID) (Conn, error) {
+	logdial := lgbl.Dial("conn", d.LocalPeer, remote, nil, raddr)
+	defer log.EventBegin(ctx, "connDial", logdial).Done()
 
 	maconn, err := d.rawConnDial(ctx, raddr, remote)
 	if err != nil {
+		logdial["dial"] = "failure"
```

Comment:
Can use named return values to check for `err` in a deferred function.

```Go
func(...) (c Conn, err error) {
defer func() {
  if err != nil {
  } 
}()
```

NB: Despite the fact that the named results would be used to enrich the log, one still would not want to use naked returns...
https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Named_Result_Parameters